packages/transport/src/chrome-runtime/background-connection-manager.ts

```diff
@@ -72,105 +77,163 @@ export class BackgroundConnectionManager {
+        const [channel] = nameChannel(ChannelSubLabel.ServerStream);
+        const acont = new AbortController();
+
         /**
-         * This is the generated connection listener. It checks for a specified
-         * name, which is transmitted to the client by the caller. Any
-         * connecting port not identifying that name is ignored.
+         * This is the generated connection listener. It checks for our
+         * generated subName, and any connection not bearing the name is
+         * ignored.  The sink transmits the stream piecewise in the channel.
          *
          * @param subPort chrome.runtime.Port, initiated by any new runtime connection
          */
-        const subConnectionListener = (subPort: chrome.runtime.Port) => {
-          if (subPort.name !== subName) return;
-          chrome.runtime.onConnect.removeListener(subConnectionListener);
-          const acont = new AbortController();
-          connection.streams.set(subName, acont);
-          subPort.onDisconnect.addListener(() => acont.abort(subName));
-          void subStream
+        const subListener = (subPort: chrome.runtime.Port) => {
+          if (subPort.name !== channel) return;
+          chrome.runtime.onConnect.removeListener(subListener);
+          connection.streams.set(channel, acont);
+          subPort.onDisconnect.addListener(() => acont.abort(channel));
+          void stream
             .pipeTo(new WritableStream(new ChromeRuntimeStreamSink(subPort)), {
               signal: acont.signal,
             })
-            .catch(e => {
-              if (e !== subName) throw e;
+            .catch((error: unknown) => {
+              if (error !== channel) {
+                // TODO: handling errors in this promise chain is problematic.
+                // we're in a callback, and can't reliably answer the requester.
+                console.error('Error in subchannel', error);
+                clientPort.postMessage({
+                  error: Object.assign(errorToJson(ConnectError.from(error), undefined), {
+                    requestId,
+                    channel,
+                  }),
+                });
+              }
             })
             .finally(() => {
-              connection.streams.delete(subName);
+              connection.streams.delete(channel);
               subPort.disconnect();
-              void subStream.cancel();
             });
         };
-        return subConnectionListener;
+        chrome.runtime.onConnect.addListener(subListener);
+
+        // TODO: revisit this
+        /*
+        setTimeout(() => {
+          if (chrome.runtime.onConnect.hasListener(subListener)) {
+            // handler still hasn't activated, or it would be gone
+            chrome.runtime.onConnect.removeListener(subListener);
+            acont.abort('Timed out subchannel activation');
+          }
+        }, 1000);
+        */
```

Comment:
Can we either delete this or give more info in the TODO as to what this is about and why it needs revisiting?